.github/workflows/presubmit.yml

```
@@ -97,11 +97,22 @@ jobs:
             build/config.log
             build/**/tests/*.log
 
-  build-install-and-integration_tests:
+  integration-tests:
     needs: [babeltrace2, pre_job]
     if: ${{ needs.pre_job.outputs.should_skip != 'true' }}
-    name: Build and Install and Integration Tests ubuntu-latest
+    name: Integration ${{ matrix.bats_file }} ${{matrix.thapi_sync_daemon }}
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        thapi_sync_daemon: [ 'fs', 'mpi' ]
+        bats_file: ['test', 'mpi_test', 'sync_daemon_test']
+        exclude:
+          - bats_file: test
+            thapi_sync_daemon: mpi
+    env:
```

```suggestion
        thapi_sync_daemon: [ 'fs', 'mpi' ]
        bats_file: ['mpi_test', 'sync_daemon_test']
        include:
          - bats_file: test

    env:
```
Maybe? Don't need to set any env for the non_mpi test in theory